### ISSUE TITLE
Docker Desktop 2.3.1.0 fails to start after upgrading from 2.3.0.1

### ISSUE THREAD

**Issue by ryanfb** [docker/for-mac#4594]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID: Diagnostics upload spins forever

### Expected behavior

Docker Desktop starts up normally after upgrading.

### Actual behavior

Docker Desktop fails to start with the message:

```
Fatal Error
Docker daemon failed to start
2020-05-21T13:22:28Z dockerd time="2020-05-21T13:22:28.439657491Z" level=debug msg="Calling GET /v1.40/version"
```

Clicking "Diagnose & Feedback" leads to a diagnostic window which simply spins forever on "Diagnosing...".

After resetting to factory defaults, Docker Desktop was able to start again.

### Information

  - macOS Version: 10.14.6

### Diagnostic logs

See above.

### Steps to reproduce the behavior

Use Docker Desktop 2.3.0.1 on a Mac. Upgrade to 2.3.1.0. Docker Desktop no longer starts.

**Comment by bpehling** [comment]
I received the same error message but a little different debug message
```
2020-05-21T13:39:25Z dockerd time="2020-05-21T13:39:25.534991167Z" level=debug msg="heartbeat successful to manager {xxxxxx 192.168.65.x:2377}, next heartbeat period: 5.245079064s" method="(*session).heartbeat" module=node/agent node.id=xxxxxx session.id=xxxxxx sessionID=xxxxxx
```

- macOS version: 10.15.3

Clicking "Diagnose & Feedback" spins indefinitely for me as well on "Diagnosing..."

**Comment by robd003** [comment]
Experiencing the same issue as @ryanfb 

Does anyone have a link to the prior release?

**Comment by djs55** [comment]
If the diagnostics in the UI are hanging, try running them from the CLI:
```
 /Applications/Docker.app/Contents/MacOS/com.docker.diagnose gather -upload
```

**Comment by jmaitrehenry** [comment]
Same issue here, my Diagnostics ID: 96A974DE-1B6E-4E3D-B4B4-F23D05DD9E22/20200521203743
The error is the same as @bpehling

**Comment by djs55** [comment]
Thanks all for your reports. I've created a ticket and added it to the backlog.

Since the error seems to be an expired certificate, it might unblock startup to move the old certs:
```
mv ~/Library/Group\ Containers/group.com.docker/pki ~/Library/Group\ Containers/group.com.docker/pki.old
```
Unfortunately this will probably mean the k8s cluster will need to be reset, but hopefully the app will start.

**Comment by jmaitrehenry** [comment]
@djs55 it works! I try to find it on `~/Library/Containers` but didn't know about the `~/Libarary/Group\ Containers` place on my mac.

**Comment by enragedginger** [comment]
My error was slightly different (the URL in my error message at startup was `_ping`). However, moving the certs like @djs55 mentioned resolved my issue. Thank you!

**Comment by tmack8001** [comment]
Removing the certs got me back into a usable state. Thanks I can confirm that on MacOS 10.15.5 this started occurring for me and after removing those certs Docker for Mac starts up.

**Comment by fdanielsen** [comment]
I can also confirm the certificate issue and that moving them resolved it for me too. Not using a Kubernetes cluster, so doesn't seem to have done any harm / need for any reset for me either. On macOS 10.15.4, and the issue occurred after upgrading to latest Edge release 2.3.1.0 (45408) today.

**Comment by hktalent** [comment]
me too
```
cannot create kubernetes PKI: error creating PKI assets: failed to write or validate certificate "apiserver": failure loading apiserver certificate: failed to load certificate: the certificate has expired
```

**Comment by ryanfb** [comment]
Fixed in 2.3.2.0: upgrading from 2.3.0.1 to 2.3.2.0 works as expected.

**Comment by stephen-turner** [comment]
As @ryanfb says, we have fixed this in the latest Edge release, 2.3.2.0.
